src/src/ESPEasyCore/ESPEasyRules.cpp

```diff
@@ -1384,29 +1384,28 @@ void createRuleEvents(struct EventStruct *event) {
 
   const byte valueCount = getValueCountForTask(event->TaskIndex);
 
-  for (byte varNr = 0; varNr < valueCount; varNr++) {
+  if (Settings.CombineTaskValues_SingleEvent(event->TaskIndex)) {
     String eventString;
-    eventString.reserve(32); // Enough for most use cases, prevent lots of memory allocations.
+    eventString.reserve(128); // Enough for most use cases, prevent lots of memory allocations.
```

Comment:
To save a few bytes of memory, reserving 64 bytes would probably be enough for most use-cases (15 character taskname + `#All=` + 4x 10 digits)

Reply:
Well the fields for task name can be 25 or 40 bytes long (not sure) plus the values, so that's "quite a lot".
I could set it to 64, but it is not going to matter a lot as the String itself is almost immediately after its creation copied to the event queue.
N.B. Copied, not moved.
So after the copy, the new String has the exact size it needs.

Also I do like to have those reserve sizes a power of 2, or at least a multiple of some power of 2 (not too small, so not '1' ;) ) to limit heap fragmentation.

The idea behind it was that allocating too much isn't a big deal (here) but allocating too little is.
So changing to 64 bytes is almost always fine too, as it will probably fit like 99% of the time.